src/genpy/base.py

```diff
@@ -72,7 +73,7 @@
     'bool': 'bool',
     'char': 'bytes',
     'time': 'genpy.Time', # See: https://github.com/Pickle-Robot/genmsg/blob/b3c24ea4a75ac494e4f94bf88e8838c71f4280a5/src/genmsg/msgs.py#L326
-    'duration': 'genpy.Time' 
+    'duration': 'genpy.Duration'
```

Comment:
What testing have you done?  I have a vague recollection of this particular line, which makes me think it's like this deliberately for some reason.  Will probably need to look into more for details